src/app/zap-templates/override.js

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+function atomicType(arg) {
+  switch (arg.name) {
+  case 'attribute_id':
+    return 'chip::AttributeId';
+  case 'cluster_id':
+    return 'chip::ClusterId';
+  default:
+    throw 'not overriding';
+  }
+}
+
+function nonAtomicType(arg) {
```

Comment:
There is an helper inside ZAP to default all struct and enum to `uint8_t *` pointer instead of overriding `nonAtomicType` it's `asUnderlyingZCLType`.

Reply:
Sadly I can't find it in the public repo :(

Reply:
The master branch was updated, so you should be able to find it now. And if I'm not mistaken all ZCL enums are of type `ENUM8` (at least from what I see in the XML files that we use)

Reply:
I removed that.